messaging/coap/coap.c

```diff
@@ -1021,17 +1021,7 @@ coap_oscore_parse_options(coap_packet_t *packet, const uint8_t *data,
       ++current_option;
       packet->payload = current_option;
       packet->payload_len = (uint32_t)(data_len - (packet->payload - data));
-
-      if (packet->transport_type == COAP_TRANSPORT_UDP &&
-          packet->payload_len > (uint32_t)OC_MAX_APP_DATA_SIZE) {
-        packet->payload_len = (uint32_t)OC_MAX_APP_DATA_SIZE;
```

Comment:
This was a weird code. I think removing it should be OK, because below we check if the options overflow the data buffer:

```C
if (current_option + option_length > data + data_len) {
```

so we should return `BAD_REQUEST_4_00` if we get a message that's too long here.